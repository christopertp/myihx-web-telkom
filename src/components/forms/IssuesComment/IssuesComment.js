import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import TextAreaField from '../../fields/TextArea';
import styles from './styles.scoped.css';

export default function IssuesComment(props) {
  const { handleSubmit, invalid } = props;
  const { message } = useSelector(s => s.issues);
  const inputProps = {
    label: 'MESSAGE/COMMENT',
    placeholder: 'Enter Message',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={TextAreaField} inputProps={inputProps} name="message" />
      {message && <small className={styles.error}>{message}</small>}
      <Button disabled={invalid} type="submit">Add Comment</Button>
    </form>
  );
}

IssuesComment.defaultProps = {
  handleSubmit: () => { },
  invalid: false,
};

IssuesComment.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
