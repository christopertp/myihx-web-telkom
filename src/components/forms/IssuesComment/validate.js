export default function validate(values) {
  const { message } = values;
  return {
    message: !message ? 'Please enter message' : '',
  };
}
