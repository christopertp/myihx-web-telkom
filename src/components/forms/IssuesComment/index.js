import { reduxForm } from 'redux-form';
import IssuesComment from './IssuesComment' ;
import validate from './validate';

export default reduxForm({
  initialValues: {
    message: '',
  },
  form: 'issuesComment',
  validate,
})(IssuesComment);
