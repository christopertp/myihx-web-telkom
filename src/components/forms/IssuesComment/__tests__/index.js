import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import IssuesComment from '../IssuesComment';

const { defaultProps } = IssuesComment;
const store = {
  message: 'error',
};

useSelector.mockImplementation(fn => {
  fn({});
  return store;
});

describe('src/components/forms/IssuesComment', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IssuesComment />);
    expect(tree).toMatchSnapshot();
    defaultProps.handleSubmit();
  });
});
