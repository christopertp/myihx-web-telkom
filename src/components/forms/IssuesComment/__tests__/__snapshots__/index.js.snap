// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/IssuesComment render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "MESSAGE/COMMENT",
        "placeholder": "Enter Message",
      }
    }
    name="message"
  />
  <small>
    error
  </small>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Add Comment
  </Button>
</form>
`;
